// @capgo/capacitor-speech-recognition/dist/plugin.cjs.js
'use strict';

var core = require('@capacitor/core');

const SpeechRecognition = core.registerPlugin('SpeechRecognition', {
    web: () => Promise.resolve().then(function () { return web; }).then((m) => new m.SpeechRecognitionWeb()),
});

class SpeechRecognitionWeb extends core.WebPlugin {
    available() {
        throw this.unimplemented('Speech recognition is not available on the web.');
    }
    async isOnDeviceRecognitionAvailable() {
        return { available: false };
    }
    start(_options) {
        throw this.unimplemented('Speech recognition is not available on the web.');
    }
    stop() {
        throw this.unimplemented('Speech recognition is not available on the web.');
    }
    forceStop(_options) {
        throw this.unimplemented('Speech recognition is not available on the web.');
    }
    async getLastPartialResult() {
        return { available: false, text: '' };
    }
    setPTTState(_options) {
        throw this.unimplemented('Speech recognition is not available on the web.');
    }
    getSupportedLanguages() {
        throw this.unimplemented('Speech recognition is not available on the web.');
    }
    isListening() {
        throw this.unimplemented('Speech recognition is not available on the web.');
    }
    checkPermissions() {
        throw this.unimplemented('Speech recognition permissions are not handled on the web.');
    }
    requestPermissions() {
        throw this.unimplemented('Speech recognition permissions are not handled on the web.');
    }
    async getPluginVersion() {
        return { version: 'web' };
    }
}

var web = /*#__PURE__*/Object.freeze({
    __proto__: null,
    SpeechRecognitionWeb: SpeechRecognitionWeb
});

exports.SpeechRecognition = SpeechRecognition;
